build-support/packer/runson/runson.pkr.hcl

```diff
@@ -0,0 +1,118 @@
+#
+# 2. Per our policy, your user must use MFA. So you must acquire temporary credentials to pass to Packer:
+#
+#    `aws --profile=pants sts get-session-token --serial-number arn:aws:iam::AWSACCOUNTID:mfa/USERNAME --token-code XXXXXX`
+#    Where AWSACCOUNTID is pantsbuild's AWS account ID, USERNAME is your username, and XXXXXX is the code
+#    generated by your MFA token.
+#
+#    This will emit temporary credentials as follows:
+#
+#    {
+#        "Credentials": {
+#            "AccessKeyId": "ACCESSKEYID",
+#            "SecretAccessKey": "SECRETACCESSKEY",
+#            "SessionToken": "SESSIONTOKEN",
+#        }
+#    }
+#
+# 3. Run Packer build with the temporary creds:
+#    ```
+#    AWS_ACCESS_KEY_ID=ACCESSKEYID AWS_SECRET_ACCESS_KEY=SECRETACCESSKEY AWS_SESSION_TOKEN=SESSIONTOKEN \
+#      packer build build-support/packer/runson/runson.pkr.hcl
+#    ```
+#
+#    This will take ~7 minutes but will eventually give you the ID of the AMI:
+#
+#    ```
+#    ==> Builds finished. The artifacts of successful builds are:
+#    --> build-custom-runson-ami.amazon-ebs.runson: AMIs were created:
+#    us-east-1: AMI-ID
+#    ```
+#
+# Updating RunsOn Config:
+#
+#  1. Create a PR that sets the `ami` field in .github/runs-on.yml to this new AMI-ID.
+#     Once this PR is merged into the main branch, RunsOn will start using the new AMI
+#     when it creates on-the-fly CI runners.
+#
+#     Note that since the Pants repo is public, RunsOn will only use config in the default
+#     branch of the repo, in our case `main`. If the reason you're updating the AMI is that
+#     CI is failing on an expired previous AMI, you will need to merge into main without green CI.
+#     This will require you to temporarily disable the branch protection rule on main.
+#
+#  2. Once the new config is in main, and any in-flight CI runs have finished, we can deregister
+#     the old AMI via the AWS web console, to avoid confusion.
+#
+# NOTE: If you edit this file, run `packer fmt build-support/packer/runson/runson.pkr.hcl` to format it
+#       and `packer validate build-support/packer/runson/runson.pkr.hcl` to validate it.
+
+packer {
+  required_plugins {
+    amazon = {
+      version = ">= 1.2.8"
+      source  = "github.com/hashicorp/amazon"
+    }
+  }
+}
+
+source "amazon-ebs" "runson" {
+  ami_name      = "ubuntu22-full-arm64-python3.7-3.13-{{timestamp}}"
+  instance_type = "t4g.nano"
+  region        = "us-east-1"
+  source_ami_filter {
+    filters = {
+      name                = "runs-on-v2.2-ubuntu22-full-arm64-*"
+      root-device-type    = "ebs"
+      virtualization-type = "hvm"
+    }
+    most_recent = true
+    owners      = ["135269210855"] # RunsOn's AWS account ID
+  }
+  # Packer creates a temporary keypair on the temporary instance it uses
+  # to create the AMI, so we don't need to configure that.
+  ssh_username = "ubuntu"
+  user_data    = "#!/bin/bash\nsystemctl start ssh"
```

Comment:
You may want to use the HCL heredoc syntax here. See https://developer.hashicorp.com/packer/docs/templates/hcl_templates/expressions#string-literals.

Specifically, the "indented heredoc" which behaves like `textwrap.dedent`: `<<~EOF`